### DISCUSSION TITLE
Repository forked to  a new home

### DISCUSSION THREAD

**Discussion by peteeckel** [auroraresearchlab/netbox-dns#299] [General]
Hi, 

as it seems, Auroraresearchlab, the maintainers of NetBox DNS, have silently left us. Unfortunately there was no notice or handover, so the only way to go on right now seems to be to fork the repo.

Since NetBox 3.5 is out and the last official version of NetBox DNS does not work with the new release, I did a quick fork and published the plugin on PyPI as `netbox-plugin-dns`. Currently there is a beta version 0.18.0b1, which should be fine for all practial purposes. Unfortunately I don't have much time right now and can't set everything up all the workflows properly, so the real 0.18.0 will have to wait until June.

I hope this quick (and slightly dirty) fix will help users who want to migrate to NetBox 3.5.

Declaring the version beta will also help moving back should Aurora come back. Currently it doesn't seem probable, though.

 Thanks, 

  Pete.

**Comment by vwbusguy** [comment]
I assume [this is the fork](https://github.com/peteeckel/netbox-plugin-dns) we'll want to track moving forward?  Also, thank you very much for doing this - both for the 3.5 compatibility and for picking this up.  It's much appreciated!

**Reply by peteeckel** [reply]
> I assume [this is the fork](https://github.com/peteeckel/netbox-plugin-dns) we'll want to track moving forward? Also, thank you very much for doing this - both for the 3.5 compatibility and for picking this up. It's much appreciated!

Yes, exactly - I took the opportunity to rename the plugin in line with many other NetBox plugins (netbox-plugin-*) is some kind of common denominator, so finding a new name was quite straightforward.

You're welcome - a couple of my customers are using NetBox DNS as well, so I have been considering this step in case something goes wrong with Auroraresearchlab for a while. It really looks like they're gone - not only the web site is down, they also don't have an IP address anymore. Pity they didn't say anything, but ... well, never mind.

**Reply by vwbusguy** [reply]
Awesome.  We're using it in my $DAYJOB, and while I'm not too keen to Netbox internals, I'm competent with Python, so please feel free to reach out if there's something I can help with along the way.

**Reply by peteeckel** [reply]
At the moment I'm particularly keen on knowing that everything works with the new package - I tried to keep it as close to the original as possible, but renaming the module might have side-effects I didn't test for or cause issues with the installation. So if everything is fine, a quick heads-up would be nice.

**Comment by peteeckel** [comment]
A quick update to this discussion. 

A bit more than one and a half year ago, I started contributing to this project because I found the issue interesting, had a need for exactly what was developed here, and like the idea of contributing to community projects in general.

Though the cooperation with the Aurora Research Lab team has never been what you might call 'cordial' (or personal in any way) it worked out fine and there was no need for any change. I did PRs, they accepted them most of the time, and from time to time they did releases and the fruit of my (largely, there were some other contributors as well but most of the code was actually written by me based on the initial releases by the Aurora guys) work was available to the public. No need to change anything, it worked. 

Until, obviously, it didn't. Aurora Research Lab disappeared without any notice, without warning anyone and without - what's worst - at least offering me or anyone else the opportunity to take over from them. No more PR merges, no more releases, NetBox DNS was dead without anyone knowing it. This is **not** what community work should look like. 

I don't know what happened, I don't blame anyone for whatever it was, it just looks like a company or a project or whatever it was was silently killed and whoever was relying on them were just left on their own. That sucks.

After thinking about it for a while I decided to take the whole project in my own hands, fork it and get over the past. Currently I'm the main contributor anyway, so it doesn't make sense to have another project owner or maintainer or whatever - for the users not much will change, except that releases may come faster in some or most cases. I won't change the license, I won't change any modalities such as contributor guidelines etc. (at least I don't plan to do so unless there's a very good reason for it), and anyone is still welcome to contribute, open issues and whatever. So, nothing changes for anyone but me. 

I sincerely thank the people who offered to fork the project so it can live on, but after making the decision to fork I don't see much sense in handing the project over and making it and my work on it dependent on anyone else again. It doesn't really have any advantages, and we just saw what can happen when whoever is responsible isn't really committed (maybe because the project is actually owned by their employer and they were paid to do it, but didn't really put their heart into it - no hard feelings, but that happens a lot).

So now it's [peteeckel/netbox-plugin-dns](https://github.com/peteeckel/netbox-plugin-dns) on GitHub, [netbox-plugin-dns](https://pypi.org/project/netbox-plugin-dns/) on PyPI. This project here is, for all I know, dead.

**Reply by jcollie** [reply]
@peteeckel thanks for all of your work on this plugin! It's been a big help for me to have this as well.
